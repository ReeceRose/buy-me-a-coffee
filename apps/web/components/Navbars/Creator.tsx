import { useWallet } from '@solana/wallet-adapter-react';
import {
  WalletDisconnectButton,
  WalletMultiButton,
} from '@solana/wallet-adapter-react-ui';
import Link from 'next/link';

import { BiHome } from 'react-icons/bi';

export const CreatorNavbar = (): JSX.Element => {
  const { connected } = useWallet();

  return (
    <nav className="fixed top-0 z-50 flex flex-wrap items-center justify-between w-full px-2 py-3 bg-transparent">
      <div className="container flex flex-wrap items-center justify-between px-4 mx-auto">
        <div className="flex items-center flex-grow shadow-none " id="nav">
          <ul className="flex flex-row ml-auto list-none">
            <li className="flex items-center">
              <Link href="/">
                <a className="mr-2 text-lg text-white leading-lg">
                  <span className="flex items-center">
                    <BiHome className="mr-2" />
                    Home
                  </span>
                </a>
              </Link>
            </li>
            <li className="flex items-center">
              {connected ? <WalletDisconnectButton /> : <WalletMultiButton />}
            </li>
          </ul>
        </div>
      </div>
    </nav>
  );
};
