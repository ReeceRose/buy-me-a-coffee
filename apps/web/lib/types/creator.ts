export type Creator = {
  username: string;
  description: string;
  bannerImageUrl: string;
  profileImageUrl: string;
  supporters: number;
  support: Support;
  wallet: string;
};

export type Support = {
  icon: string;
  item: string;
  defaults: number[];
  pricePerItem: number;
};
