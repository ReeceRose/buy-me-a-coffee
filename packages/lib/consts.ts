export const MAINNET = 'mainnet';
export const TESTNET = 'testnet';
export const DEVNET = 'devnet';
