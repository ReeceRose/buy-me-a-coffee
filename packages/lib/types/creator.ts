export type Creator = {
  username: string;
  description: string;
  bannerImageUrl: string;
  profileImageUrl: string;
};
