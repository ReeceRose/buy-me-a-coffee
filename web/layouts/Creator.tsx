import SEO from '@/components/SEO';
import CreatorNavbar from '@/components/Navbars/CreatorNavbar';

interface Props {
  children: React.ReactNode;
  title: string;
  description: string;
}

export default function CreatorLayout({
  children,
  title,
  description,
}: Props): JSX.Element {
  return (
    <>
      <SEO title={title} description={description} />
      <main>
        <CreatorNavbar />
        {children}
      </main>
    </>
  );
}
