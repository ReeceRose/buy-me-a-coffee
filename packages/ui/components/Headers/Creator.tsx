import Image from 'next/image';

import { CreatorProps } from 'lib/types/props';

export const CreatorHeader = ({ creator }: CreatorProps): JSX.Element => {
  return (
    <>
      <div className="w-full px-6 m-auto text-center">
        <div className="relative h-64 overflow-hidden bg-gray-500 rounded-lg">
          <div className="absolute top-0 right-0 block w-full h-full">
            <Image
              className="object-cover h-full min-w-full"
              layout="fill"
              src={creator.bannerImageUrl}
            />
          </div>
        </div>
      </div>
      <div className="relative flex flex-col w-full mt-16 mb-6 break-words">
        <div className="px-6">
          <div className="flex flex-wrap justify-center">
            <div className="flex justify-center w-full px-4">
              <div className="relative overflow-hidden border-8 border-white rounded-full w-36 h-36 -m-44">
                <Image
                  alt={creator.username}
                  objectFit="cover"
                  layout="fill"
                  src={creator.profileImageUrl}
                  className="absolute align-middle"
                />
              </div>
            </div>
            <div className="w-full px-4 -mt-10 text-center">
              <div className="flex justify-center py-4 pt-8 lg:pt-4">
                <div className="p-3 text-center">
                  <span className="block text-3xl font-bold tracking-wide text-blueGray-600">
                    {creator.username}
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </>
  );
};
