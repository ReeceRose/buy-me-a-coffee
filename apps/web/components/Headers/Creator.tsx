import Image from 'next/image';

import { CreatorProps } from '@/lib/props/';

export const CreatorHeader = ({ creator }: CreatorProps): JSX.Element => {
  return (
    <>
      <div className="w-full px-6 m-auto text-center ">
        <div className="relative h-64 overflow-hidden bg-gray-500 rounded-lg">
          <div className="absolute top-0 right-0 block w-full h-full">
            <Image
              className="object-cover h-full min-w-full"
              layout="fill"
              alt="Creator banner image"
              src={creator.bannerImageUrl}
            />
          </div>
        </div>
      </div>
      <div className="relative flex flex-col w-full mt-16 mb-6 break-words">
        <div className="px-6">
          <div className="flex flex-wrap justify-center">
            <div className="flex justify-center w-full px-4">
              <div className="relative overflow-hidden border-8 border-white rounded-full w-36 h-36 -m-44">
                <Image
                  alt={creator.username}
                  objectFit="cover"
                  layout="fill"
                  src={creator.profileImageUrl}
                  className="absolute align-middle bg-gray-500"
                />
              </div>
            </div>
            <div className="w-full px-4 -mt-12 text-center ">
              <div className="flex justify-center pt-8 lg:pt-4">
                <div className="text-center ">
                  <span className="block text-3xl font-bold tracking-wide">
                    {creator.username}
                  </span>
                  <span className="block mt-1 text-xl tracking-wide">
                    {creator.description}
                  </span>
                  <span className="block mt-5 text-lg tracking-wide text-gray-500">
                    {creator.supporters} supporters
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </>
  );
};
