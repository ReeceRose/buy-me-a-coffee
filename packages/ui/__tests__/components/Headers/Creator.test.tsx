import { render } from '@testing-library/react';
import { Creator } from 'lib/types/';

import { CreatorHeader } from '../../../components/Headers/Creator';

describe('Creator headbar tests', () => {
  const creator: Creator = {
    bannerImageUrl:
      'https://www.pngitem.com/pimgs/m/505-5058955_sample-png-images-sample-png-transparent-png.png',
    profileImageUrl:
      'https://www.pngitem.com/pimgs/m/505-5058955_sample-png-images-sample-png-transparent-png.png',
    username: 'TestUser',
    description: 'Test user is helping with tests',
  };

  test('Headbar matches snapshot', () => {
    const { asFragment } = render(<CreatorHeader creator={creator} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
