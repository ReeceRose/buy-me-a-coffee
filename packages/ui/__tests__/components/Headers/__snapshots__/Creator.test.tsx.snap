// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creator headbar tests Headbar matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="w-full px-6 m-auto text-center"
  >
    <div
      class="relative h-64 overflow-hidden bg-gray-500 rounded-lg"
    >
      <div
        class="absolute top-0 right-0 block w-full h-full"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            class="object-cover h-full min-w-full"
            data-nimg="fill"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
    </div>
  </div>
  <div
    class="relative flex flex-col w-full mt-16 mb-6 break-words"
  >
    <div
      class="px-6"
    >
      <div
        class="flex flex-wrap justify-center"
      >
        <div
          class="flex justify-center w-full px-4"
        >
          <div
            class="relative overflow-hidden border-8 border-white rounded-full w-36 h-36 -m-44"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="TestUser"
                class="absolute align-middle"
                data-nimg="fill"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
              />
              <noscript />
            </span>
          </div>
        </div>
        <div
          class="w-full px-4 -mt-10 text-center"
        >
          <div
            class="flex justify-center py-4 pt-8 lg:pt-4"
          >
            <div
              class="p-3 text-center"
            >
              <span
                class="block text-3xl font-bold tracking-wide text-blueGray-600"
              >
                TestUser
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
